File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/Cultist_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class Cultist_Model extends MonsterInfo{
    public static Id = "Cultist"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(50,56);
        else
            return Mathf.RandomRange(48,54);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/cultist/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/cultist/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/cultist/skeleton.png"));
    }
    
    get Name(): string {
        return "异教徒";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/Cultist_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class Cultist_Model : MonsterInfo
    {
        public static string Id = "Cultist";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(50,56) : Random.Range(48,54);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/cultist/skeleton.atlas",
                    "Monsters/theBottom/cultist/skeleton.json", "Monsters/theBottom/cultist/skeleton.png");
            }
        }

        public override string Name => "异教徒";

        public Cultist_Model(Vector2 pos) : base(pos)
        {
        }
    }
}